docs/codeactions/code-actions.md

```diff
@@ -0,0 +1,96 @@
+
+Here is a list of the code actions supported by Metals. Some of these code actions support much more complex use cases, which you would eventually discover through your code, by clicking on the magic 💡, as it appears, to get your surprise!
+
+## Convert To Named Arguments
+
+This code action adds the name to parameters passed to a method call or a class instant.
+
+[Convert To Named Arguments Code Action](https://github.com/scalameta/metals/docs/codeactions/gifs/ConvertToNamedArguments.gif)
+
+
+## Create Companion Object
+
+If you have a `trait` or a `class` definition, you can invoke this code action for creating its companion object.
+
+[Create Companion Object Code Action](https://github.com/scalameta/metals/docs/codeactions/gifs/CreateCompanionObject.gif)
+
+## Create New Symbol
+
+This is for creating a definition for a symbol which is used in your code, but not defined anywhere.
+
+[Create New Symbol Code Action](https://github.com/scalameta/metals/docs/codeactions/gifs/CreateNewSymbol.gif)
+
+## Extract Member
+
+Extract a class, trait, or object definition to a new file.
+
+[Extract Member Code Action](https://github.com/scalameta/metals/docs/codeactions/gifs/ExtractMember.gif)
+
+## Rename File
+
+Renames a file to match that of the prominent type defined in it.
+
+[Rename File](https://github.com/scalameta/metals/docs/codeactions/gifs/RenameFile.gif)
+
+### Extract Value
+
+If you think your code would look nicer, if the calcuation of a value is extracted and assined to a new `val`; this code action would come to your aid.
+[Extract Value](https://github.com/scalameta/metals/docs/codeactions/gifs/ExtractValue.gif)
+
+### map, flatMap, withFilter to For Yield
+
+It converts a chain of `map`, `flatMap`, `filter` and `filterNot` methods into a for comprehension
+
+[To For Comprehension](https://github.com/scalameta/metals/docs/codeactions/gifs/FlatMapToForComprehension.gif)
+
+## Implement Abstract Members of the Parent Type
+
+Upon inherting from a type, you also have to implement its abstract members. But manuallylooking them all up and copying their signature is time consuming, isn't it? You can just use this code action instead.
+
+[Implement Abstract Members](https://github.com/scalameta/metals/docs/codeactions/gifs/ImplementAbstractMembers.gif)
+
+### Import Missing Symbol 
+
+Because, of course, you often do not remember the package where it is located, without using this code action. right?
+
+[Import Missing Symbol](https://github.com/scalameta/metals/docs/codeactions/gifs/ImportMissingSymbol.gif)
+
+## Insert Inferred Type
+
+This one, tries to infer the type of variables according to the context, and insert them for you.
+
+[Insert Inferred Type](https://github.com/scalameta/metals/docs/codeactions/gifs/InsertInferredType.gif)
+
+## Organize Imports
+
+Yes, just have all the clutter, as well as the unnecessary ones removed and tidied up.
```

Comment:
This is a bit unclear, it would also be good to mention that we need a special compiler option for removing unused imports and that in some editors it's not available on the "lgiht bulb"